core/trino-main/src/main/java/io/trino/util/SystemUtils.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.util;
+
+import com.google.common.base.StandardSystemProperty;
+import oshi.SystemInfo;
+
+import static java.lang.Math.min;
+
+public final class SystemUtils
```

Comment:
"XXXUtils" is not a great name. How about just calling it `MachineInfo`, `SystemInfo`, or something along those lines?

Reply:
renamed to `MachineInfo`